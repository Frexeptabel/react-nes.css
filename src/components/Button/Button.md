Button example:

```js
<Button>This is a button</Button>
<Button color="primary">This is a button</Button>
<Button color="success">This is a button</Button>
<Button color="warning">This is a button</Button>
<Button color="error">This is a button</Button>
<Button color="disabled">This is a button</Button>
```
