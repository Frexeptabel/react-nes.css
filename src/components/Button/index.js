import React from "react";
import PropTypes from "prop-types";
import classNames from "classnames";

const Button = ({ color, ...props }) => {
  const className = classNames("nes-btn", color && `is-${color}`);
  return (
    <button className={className} {...props}>
      {props.children}
    </button>
  );
};

Button.propTypes = {
  color: PropTypes.oneOf(["primary", "success", "warning", "error", "disabled"])
};

export default Button;
