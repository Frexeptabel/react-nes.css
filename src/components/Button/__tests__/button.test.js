import React from "react";
import Button from "../index";
import renderer from "react-test-renderer";

test("The Button should be rendered properly", () => {
  const component = renderer.create(<Button>Test</Button>);
  expect(component).toMatchSnapshot();
});

test("The Button should take the color property", () => {
  let component = renderer.create(<Button color="success">Test</Button>);
  expect(component).toMatchSnapshot();

  component = renderer.create(<Button color="warning">Test</Button>);
  expect(component).toMatchSnapshot();

  component = renderer.create(<Button color="success">Test</Button>);
  expect(component).toMatchSnapshot();

  component = renderer.create(<Button color="primary">Test</Button>);
  expect(component).toMatchSnapshot();

  component = renderer.create(<Button color="error">Test</Button>);
  expect(component).toMatchSnapshot();

  component = renderer.create(<Button color="disabled">Test</Button>);
  expect(component).toMatchSnapshot();
});
