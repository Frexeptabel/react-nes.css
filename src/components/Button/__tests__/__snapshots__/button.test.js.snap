// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Button should be rendered properly 1`] = `
<button
  className="nes-btn"
>
  Test
</button>
`;

exports[`The Button should take the color property 1`] = `
<button
  className="nes-btn is-success"
>
  Test
</button>
`;

exports[`The Button should take the color property 2`] = `
<button
  className="nes-btn is-warning"
>
  Test
</button>
`;

exports[`The Button should take the color property 3`] = `
<button
  className="nes-btn is-success"
>
  Test
</button>
`;

exports[`The Button should take the color property 4`] = `
<button
  className="nes-btn is-primary"
>
  Test
</button>
`;

exports[`The Button should take the color property 5`] = `
<button
  className="nes-btn is-error"
>
  Test
</button>
`;

exports[`The Button should take the color property 6`] = `
<button
  className="nes-btn is-disabled"
>
  Test
</button>
`;
